### PR TITLE
Add breathing room to seek back between control points in editor

### LINKED ISSUE #28885: Getting stuck while navigating the timeline backwards using the up arrow key during active playback
### Type

Game behaviour

### Bug description

I discovered that you can navigate a difficulty of a beatmap in the editor using up and down arrow keys, but when I wanted to compare two kiai sections that I thought sounded similar, I found it difficult to navigate backwards because it kept snapping to the same timing. I can navigate by pressing left arrow key then up arrow key or by pausing, but I think a different solution is preferable.

### Screenshots or videos

https://github.com/user-attachments/assets/3cce041d-2362-4a49-8ef8-4182fa467da7

### Version

2024.625.2

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16252296/compressed-logs.zip)

### LINKED ISSUE #30070: Seeking between timing sections in editor doesn't match stable.
### Type

Game behaviour

### Bug description

I've started mapping in lazer and this is probably the biggest gripe I have so far. In stable `Up` seeks forward to the nearest timing section while `Down` seeks backwards. This is inversed in Lazer breaking all muscle memory. Backwards seek leniency also doesn't exist in Lazer making it a pain to seek backwards as you effectively can't do it while the song is playing.

### Screenshots or videos

_No response_

### Version

2024.906.2-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17200300/compressed-logs.zip)

### PR DESCRIPTION
**Pull request by TaterToes**
closes #28885 (and #30070)

gives 450 ms to seek back after last control point in editor.

### CODE REVIEW

**Comment by bdach** [comment]
Not anything you need to concern yourself with. That problem is already fixed in master.

**Comment by bdach** [comment]
Stable logic for this is

```csharp
            if (AudioEngine.ActiveInheritedTimingPointIndex < 0) return;

            if (AudioEngine.Time - AudioEngine.ControlPoints[AudioEngine.ActiveInheritedTimingPointIndex].Offset > 1000)
            {
                AudioEngine.SeekTo((int)AudioEngine.ControlPoints[AudioEngine.ActiveInheritedTimingPointIndex].Offset);
                return;
            }

            if (AudioEngine.ActiveInheritedTimingPointIndex == 0)
                return;

            AudioEngine.SeekTo((int)AudioEngine.ControlPoints[AudioEngine.ActiveInheritedTimingPointIndex - 1].Offset);
```

(https://github.com/peppy/osu-stable-reference/blob/38bbc10a53c78d83c9f04a04f7bf4818505cd313/osu!/GameModes/Edit/Editor.cs#L522-L536)

In particular note:

- the different lenience interval (1000ms) - I'd probably apply this change
- the *lack* of dispensation for when the track is running, which is present in this PR - I think checking track running state is fine, so I wouldn't want any changes with respect to this

(Side note but the stable implementation is bugged - if you have a non-inherited point and inherited point on the same timestamp, e.g. a "yellow line", you won't be able to seek back beyond it. Not relevant in this context anyhow.)

@peppy obligatory summon for UX opinion on desired behaviour

File: osu.Game/Screens/Edit/Editor.cs [hunk 1]
```diff
@@ -1098,8 +1098,17 @@ private void updateSampleDisabledState()
 
         private void seekControlPoint(int direction)
         {
+            // Gives margin to seek back after last control point
+            double seekMargin = 0;
+            
+            if (clock.IsRunning) 
+            {
+                IAdjustableClock adjustableClock = clock;
+                seekMargin = 450 * adjustableClock.Rate;
```
Comment (peppy) [inline osu.Game/Screens/Edit/Editor.cs:1107]: I don't think rate needs to be accounted for? Especially if we increase this to 1 second.

Comment (peppy) [reply osu.Game/Screens/Edit/Editor.cs:1107]: Probably fine to leave it, whatever works.

**Review by peppy — APPROVED** [review]

**Comment by bdach** [comment]
> I really do question 8 commits for 10 LoC changes and it's still failing basic formatting requirements, but let's overlook that for now 🙈

I've given up on even remarking on that sort of thing at this point because it helps nothing and only looks snarky.

**Review by bdach — APPROVED** [review]

**Comment by peppy** [comment]
I guess in an attempt to make it non-snarky, @TaterToes consider not PRing until you're done with making changes, then squash your changes down to less commits if you want to seem more competent 😅

**Pull request merged**